### PR TITLE
feat: validate sufficient funds on all transaction forms

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
Shows insufficient funds error in amount/fee inputs before review steps, preventing user to move forward.

 Forms included:
 * `SendEntityRegistration`
 * `SendEntityUpdate`
 * `SendEntityResignation`
 * `SendTransfer`
 * `SendVote`
 * `SendIpfs`

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by faustbrian** [comment]
Task linked: [CU-8wx4ne ALL Transactions: Check that the wallet has enough balance to send the transaction](https://app.clickup.com/t/8wx4ne)

**Comment by goga-m** [comment]
@faustbrian making it a draft to adjust the error message UI as:

Captura de Tela 2020-10-09 a_s 18 21 14

File: src/domains/transaction/utils.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export const evaluateFee = (fee: any) => {
 };
 
 export const hasSufficientFunds = ({ wallet, fee = 0, amount = 0 }: SufficientFundsParams) => {
+	if (wallet.balance().isEqualTo(0)) return false;
```
Comment (dated) [inline src/domains/transaction/utils.ts:17]:
```suggestion
	if (wallet.balance().isZero()) return false;
```
There's a helper method for checks against zero, as zero can be either positive or negative. I think `isEqualTo(0)` just check against `+0`, haven't tested though to be honest and I don't know under what circumstances zero becomes negative 🤷

**Review by luciorubeens — COMMENTED** [review]
Also needs to be added to all transaction forms

**Comment by goga-m** [comment]
@luciorubeens yeah basically, tI did it like that, because the `AddRecipient` component has its own useForm inside the main form (that contains the amount per recipient (single/multiple), and it's logic is binded with the recipients (recipients * amount of each recipient), so I needed it to update the main form and do the calculation on that level after the recipients were added. 

This needs to be changed, indeed.  `AddRecipient` component needs refactoring to useFormContext and to propagate the change events properly. There is also an 'Add recipient' button on multiple tab which also needs to play well with the validation. We had this conversation with @samharperpittam yesterday and I started working on it yesterday to adjust the required UX, as well as rewrite some tests regarding `AddRecipient` component.

Marking this PR as draft until I finish up the refactoring :+1:

**Comment by goga-m** [comment]
Closing this in favor of https://github.com/ArkEcosystem/desktop-wallet/pull/2992

**Pull request closed without merging**